mp2-v1/tests/common/cases/table_source.rs

```diff
@@ -1426,151 +700,265 @@ impl SingleStructExtractionArgs {
         }]
     }
 
-    pub async fn generate_extraction_proof_inputs(
-        &self,
+    fn secondary_index_identifier(&self, contract: &Contract) -> Option<u64> {
+        self.secondary_index_slot_input().map(|slot_input| {
+            identifier_for_value_column(&slot_input, &contract.address, contract.chain_id, vec![])
+        })
+    }
+
+    fn metadata(&self, contract: &Contract) -> Vec<MetadataGadget> {
+        let table_info = table_info(contract, self.slot_inputs.clone());
+        metadata(&table_info)
+    }
+
+    fn storage_slots(&self, metadata: &[MetadataGadget]) -> Vec<StorageSlot> {
```

Comment:
I know it's not related to this PR but it's frankly confusing to see an Array of metadata gadget.
* Gadget name refers usually to circuit, here we're not dealing with any circuits
* The reason there's an array is just to separate the columns that corresponds to a certain EVM word. But a regular `HashMap` works as well. We could have a single struct `Metadata {  words: HashMap<EvmWord, MetadataSubset>` for example.

and that would allow us to avoid doing weird shenanigans with re-ordering the columns according to the EVM words being used ([here](https://github.com/Lagrange-Labs/mapreduce-plonky2/blob/b33cd73122f8eed3acc6625438cd6962149e30f7/mp2-v1/src/values_extraction/gadgets/metadata_gadget.rs#L63-L65)) etc.

Reply:
I also don't find this super intuitive for the end user. Maybe we could change mp2-v1 APIs (i.e., like [these methods](https://github.com/Lagrange-Labs/mapreduce-plonky2/blob/generic-extraction-integration-test/mp2-v1/src/values_extraction/api.rs#L82)) to take as input, in place of `ColumnsMetadata`, the description of the table (i.e., `Vec<ColumnInfo>`) and the set of ids of columns to be extracted, so that `MetadataGadget` (which is a circuit specific data structure, and so should not be directly exposed if possible) can always be constructed inside `mp2-v1`? Alternatively, in place of the set of ids of columns to be extracted, we might only provide the `evm_word` as input and then the columns to be extracted are the ones with `ColumnInfo.evm_word == evm_word`?
This would also allow to avoid having both `MetadataGagdet` and the "alias" struct `ColumnsMetadata` being publicly exposed, which looks a bit weird to me. Wdyt?